### PR TITLE
pythonPackages.isort: 4.2.5 -> 4.3.4

### PR DESCRIPTION
**Pull request by couchemar**
###### Motivation for this change

Update isort to the most recent version

Related to #35444
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -8977,13 +8977,13 @@ in {
   isort = buildPythonPackage rec {
     name = "${pname}-${version}";
```
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix]: drop this line

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -8977,13 +8977,13 @@ in {
...
     pname = "isort";
-    version = "4.2.5";
+    version = "4.3.4";
     src = pkgs.fetchurl {
```
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix]: Please use `fetchPypi` instead

**Review by dotlambda — CHANGES_REQUESTED** [review]
Would you mind moving the expression out of `python-packages.nix`?

**Comment by couchemar** [comment]
I think isort is a dependency of the some other packages or apps, and we cannot easy move it out of the `python-packages.nix`

**Comment by FRidh** [comment]
@couchemar the expression can be moved to `pkgs/development/python-modules/isort/default.nix` and then then called from `python-packages.nix` with `callPackage`.

**Comment by couchemar** [comment]
ok, i will change soon

**Comment by couchemar** [comment]
@dotlambda @FRidh 
Expression is in separate file now.

**Review by dotlambda — COMMENTED** [review]
Please add some blank lines where appropriate

File: pkgs/development/python-modules/isort/default.nix [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+buildPythonPackage rec {
+  pname = "isort";
+  version = "4.3.4";
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1y0yfv56cqyh9wyg7kxxv9y5wmfgcq18n7a49mp7xmzka2bhxi5r";
+  };
+  propagatedBuildInputs = [ ] ++ optionals isPy27 [ futures ];
+  # Some tests not pass
```
[the remark this answers is not in the document]
Comment (couchemar) [reply pkgs/development/python-modules/isort/default.nix]: No, there is 3 tests which fails when i make `nox-review ...` I have no idea why...

**Comment by dotlambda** [comment]
@couchemar I simply disabled the failing tests. Please have a look.

@GrahamcOfBorg build python2.pkgs.isort python3.pkgs.isort

**Comment by couchemar** [comment]
Is there reasons do not merge this request?

**Comment by dotlambda** [comment]
None besides me not noticing your approval.

**Pull request merged**